engine/function/src/templates/Basic.ftl

```diff
@@ -842,6 +842,32 @@ public class Basic {
         return ifelseObj(new BooleanVectorDirect(condition), trueCase, falseCase);
     }
 
+    /**
+     * Returns an array with null values replaced with preceding non-null value
```

Comment:
Suggested wording, cribbed a little bit from Arrays.copyOf (any thoughts, Chip?)

Copies the specified array, replacing elements that represent null in the Deephaven convention by the most recently encountered non-null value if one exists. Otherwise (if no such value exists), replaces those elements with null.

[note also that there's always this issue where the description and the @return clause are somewhat duplicative. I chose to put more detail in the description, whereas you chose to put more detail in the @return clause. I think I like it "my way" better? Maybe Chip has an opinion]